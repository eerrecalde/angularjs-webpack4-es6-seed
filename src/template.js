// file automatically generated by angular-template-cache

import angular from 'angular';

const templatesModule = angular
	.module('templates', [])
	.run(['$templateCache', function($templateCache) {
		$templateCache.put('components/breadcrumbs/breadcrumbs.tpl.html', '<nav aria-label="breadcrumb">\n' +
			'  <ol class="mt-3 mb-4 bg-secondary breadcrumb breadcrumb-border">\n' +
			'    <li ng-repeat="crumb in breadcrumbs" ng-class="[\'breadcrumb-item\', { active: $last }]" aria-current="{{!$last || \'page\'}}">\n' +
			'      <a ng-if="!$last" ng-href="{{ crumb.href }}">{{ crumb.displayName }}</a>\n' +
			'      <span ng-show="$last">{{ crumb.displayName }}</span>\n' +
			'    </li>\n' +
			'  </ol>\n' +
			'</nav>');

		$templateCache.put('components/footer/footer.tpl.html', '<footer class="ctp-footer" ng-click="footerClick()">\n' +
			'  <div class="container mt-5">\n' +
			'    <div class="row mb-3 pt-5">\n' +
			'      <div class="col-lg-2" ng-hide="category.length > 0">\n' +
			'        <h5 class="ctp-footer__heading">Categories</h5>\n' +
			'        <ul class="ctp-footer__links list-links list-unstyled">\n' +
			'          <li ng-repeat="category in categories">\n' +
			'              <a ng-href="/#/catalog/?categoryId={{category.itemId}}">\n' +
			'                 {{ category.itemName }}\n' +
			'              </a> \n' +
			'          </li>\n' +
			'        </ul>\n' +
			'      </div>\n' +
			'\n' +
			'      <div class="col-lg-2" ng-show="category.length > 0">\n' +
			'        <h5 class="ctp-footer__heading">More Links</h5>\n' +
			'        <ul class="ctp-footer__links list-links list-unstyled">\n' +
			'          <li><a href="https://www.techsoup.org/#/1">Link 1</a></li>\n' +
			'          <li><a href="https://www.techsoup.org/#/2">Link 2</a></li>\n' +
			'          <li><a href="https://www.techsoup.org/#/3">Link 3</a></li>\n' +
			'          <li><a href="https://www.techsoup.org/#/4">Link 4</a></li>\n' +
			'          <li><a href="https://www.techsoup.org/#/5">Link 5</a></li>\n' +
			'          <li><a href="https://www.techsoup.org/#/6">Link 6</a></li>\n' +
			'        </ul>\n' +
			'      </div>\n' +
			'      \n' +
			'      <div class="col-lg-2">\n' +
			'        <h5 class="ctp-footer__heading">MORE TECHSOUP</h5>\n' +
			'        <ul class="ctp-footer__links list-links list-unstyled">\n' +
			'          <li><a target="_blank" href="https://www.techsoup.org/about-us">About Us</a></li>\n' +
			'          <li><a target="_blank" href="https://www.techsoup.org/about-us/what-we-do">Our Mission</a></li>\n' +
			'          <li><a target="_blank" href="http://meet.techsoup.org/about-us/techsoup-global-network">TechSoup Global Network</a></li>\n' +
			'          <li><a target="_blank" href="https://www.techsoup.org/community/success-stories">Testimonials</a></li>\n' +
			'          <li><a target="_blank" href="https://www.techsoup.org/about-us/about-our-donor-partners">Meet Our Donor Partners</a></li>\n' +
			'          <li><a target="_blank" href="http://meet.techsoup.org/about-us/funders-and-donors">Meet Our Funders</a></li>\n' +
			'          <li><a target="_blank" href="https://www.techsoup.org/anti-discrimination-policy">Anti-Discrimination Policy</a></li>\n' +
			'        </ul>\n' +
			'      </div>\n' +
			'\n' +
			'      <div class="col-lg-2">\n' +
			'        <h5 class="ctp-footer__heading">GET IN TOUCH</h5>\n' +
			'        <ul class="ctp-footer__links list-links list-unstyled">\n' +
			'          <li><a target="_blank" href="https://npo.networkforgood.org/Donate/Donate.aspx?npoSubscriptionId=3244">Donate</a></li>\n' +
			'          <li><a target="_blank" href="https://www.techsoup.org/joining-techsoup/become-a-donor-partner">Become a Donor Partner</a></li>\n' +
			'          <li><a target="_blank" href="https://www.techsoup.org/joining-techsoup/become-a-volunteer">Volunteer</a></li>\n' +
			'          <li><a target="_blank" href="http://meet.techsoup.org/get-involved/careers">Careers</a></li>\n' +
			'          <li><a target="_blank" href="https://www.techsoup.org/about-us/contact-us">Contact Us</a></li>\n' +
			'          <li><a target="_blank" href="https://www.techsoup.org/returns-and-refunds">Returns and Refunds</a></li>\n' +
			'          <li><a target="_blank" href="http://meet.techsoup.org/about-us/media-and-press">Media and Press</a></li>\n' +
			'        </ul>\n' +
			'      </div>\n' +
			'\n' +
			'      <div class="col-lg-2">\n' +
			'        <h5 class="ctp-footer__heading">Follow Us</h5>\n' +
			'        <socials></socials>\n' +
			'      </div>\n' +
			'\n' +
			'      <!-- Begin Newsletter -->\n' +
			'      <div class="col-lg-4">\n' +
			'        <h5 class="ctp-footer__heading">Subscribe to our newsletters</h5>\n' +
			'        <div>\n' +
			'          <p>Get technology news and updates on exciting new offers from TechSoup.</p>\n' +
			'          <a target="_blank" href="https://www.techsoup.org/subscribe-to-our-free-newsletters">        \n' +
			'            <button class="btn btn-primary">SUBSCRIBE</button>\n' +
			'          </a>  \n' +
			'        </div>\n' +
			'        <!-- End Newsletter-->\n' +
			'      </div>\n' +
			'    </div>\n' +
			'\n' +
			'  <div class="col-lg-5 p-0 pb-3">\n' +
			'    <div class="d-md-inline align-middle">\n' +
			'      <i class="fa fa-phone text-secondary"></i> \n' +
			'      <span class="text-muted"><a href="tel:008006593579">(+800)-659-3579</a></span>\n' +
			'    </div>\n' +
			'    <div class="d-md-inline ml-md-4 align-middle">\n' +
			'      <i class="text-secondary fa fa-envelope-o"></i>\n' +
			'      <span class="text-muted"><a href="mailto:customerservice@techsoup.org">customerservice@techsoup.org</a></span>\n' +
			'    </div>\n' +
			'  </div>\n' +
			'</div>\n' +
			'\n' +
			'\n' +
			'  <!-- Begin  Footer Copyright -->\n' +
			'  <div class="ctp-footer__copyright bg-light">\n' +
			'    <div class="text-center">\n' +
			'      <ul class="list-inline mb-0 py-2">\n' +
			'        <li class="list-inline-item copyright">\n' +
			'          Â©All Rights Reserved\n' +
			'        </li>\n' +
			'        <li class="ctp-footer__links list-inline-item link-item">\n' +
			'          <a href="">Privacy Policy</a>\n' +
			'        </li>\n' +
			'        <li class="ctp-footer__links list-inline-item link-item">\n' +
			'          <a href="">Terms of Use</a>\n' +
			'        </li>\n' +
			'      </ul>\n' +
			'    </div>\n' +
			'  </div>\n' +
			'\n' +
			'  <!-- End Footer Copyright -->\n' +
			'</footer>');

		$templateCache.put('index.html', '<!DOCTYPE html>\n' +
			'<html>\n' +
			'<head lang="en">\n' +
			'  <meta charset="utf-8">\n' +
			'  <meta http-equiv="X-UA-Compatible" content="IE=edge">\n' +
			'  <title>My App</title>\n' +
			'  <link rel="stylesheet" href="">\n' +
			'</head>\n' +
			'<body ng-app="app">\n' +
			'\n' +
			'  <script src="bundle.js"></script>\n' +
			'\n' +
			'  <!-- Router Templates Render Here  -->\n' +
			'  <div class="container">\n' +
			'    <breadcrumbs breadcrumb-list="[{\'displayName\':\'Test 1\',\'href\':\'http://techsoup.org/\'},{\'displayName\':\'Test 2\',\'href\':\'http://techsoup.org/\'}]"></breadcrumbs>\n' +
			'    <h2>Footer below</h2>\n' +
			'    <ctp-ecom-footer></ctp-ecom-footer>\n' +
			'  </div>\n' +
			'\n' +
			'</body>\n' +
			'</html>');
	}
]);

export default templatesModule;
