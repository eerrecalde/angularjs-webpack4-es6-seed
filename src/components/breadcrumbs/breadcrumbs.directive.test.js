import './index';
import '../../template'; // eslint-disable-line import/no-unresolved

describe('Directive - Breadcrumbs', () => {
  // List of modules to include
  const modules = ['templates', 'breadcrumbsModule'];
  modules.forEach((el) => {
    beforeEach(angular.mock.module(el)); // eslint-disable-line
  });

  let compile;
  let scope;

  const injectModules = (_$compile_, _$rootScope_) => {
    const rootScope = _$rootScope_;
    compile = _$compile_;
    scope = rootScope.$new();
  };

  function getCompiledElement(lst) {
    scope.breadcrumbList = lst || [{ displayName: 'Home' }];
    let element = '<breadcrumbs breadcrumb-list="breadcrumbList"></breadcrumbs>';
    element = compile(element)(scope);
    scope.$digest();
    return element;
  }

  beforeEach(() => {
    inject(injectModules); // eslint-disable-line
  });

  it('Component by default should render one breadcrumb with text Home', () => {
    const component = getCompiledElement();
    const txt = component.text().replace(/\n|\r/g, ' ').replace(/\s\s+/g, ' ').trim();
    expect(component.html()).toMatchSnapshot();
    expect(txt).toBe('Home');
  });

  it('Creating a whole new breadcrumb with Test 1 and Test 2 should render both', () => {
    const component = getCompiledElement([{ displayName: 'Test 1', href: 'http://google.com/' }, { displayName: 'Test 2', href: 'http://google.com/' }]);
    const txt = component.text().replace(/\n|\r/g, ' ').replace(/\s\s+/g, ' ').trim();
    expect(component.html()).toMatchSnapshot();
    expect(txt).toBe('Test 1 Test 1 Test 2');
  });
});
