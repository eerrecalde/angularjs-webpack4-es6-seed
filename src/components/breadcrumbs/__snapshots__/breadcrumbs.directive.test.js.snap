// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directive - Breadcrumbs Component by default should render one breadcrumb with text Home 1`] = `
"<nav aria-label=\\"breadcrumb\\">
  <ol class=\\"mt-3 mb-4 bg-secondary breadcrumb breadcrumb-border\\">
    <!-- ngRepeat: crumb in breadcrumbs --><li ng-repeat=\\"crumb in breadcrumbs\\" ng-class=\\"['breadcrumb-item', { active: $last }]\\" aria-current=\\"page\\" class=\\"ng-scope breadcrumb-item active\\">
      <!-- ngIf: !$last -->
      <span ng-show=\\"$last\\" class=\\"ng-binding\\">Home</span>
    </li><!-- end ngRepeat: crumb in breadcrumbs -->
  </ol>
</nav>"
`;

exports[`Directive - Breadcrumbs Creating a whole new breadcrumb with Test 1 and Test 2 should render both 1`] = `
"<nav aria-label=\\"breadcrumb\\">
  <ol class=\\"mt-3 mb-4 bg-secondary breadcrumb breadcrumb-border\\">
    <!-- ngRepeat: crumb in breadcrumbs --><li ng-repeat=\\"crumb in breadcrumbs\\" ng-class=\\"['breadcrumb-item', { active: $last }]\\" aria-current=\\"true\\" class=\\"ng-scope breadcrumb-item\\">
      <!-- ngIf: !$last --><a ng-if=\\"!$last\\" ng-href=\\"http://techsoup.org/\\" class=\\"ng-binding ng-scope\\" href=\\"http://techsoup.org/\\">Test 1</a><!-- end ngIf: !$last -->
      <span ng-show=\\"$last\\" class=\\"ng-binding ng-hide\\">Test 1</span>
    </li><!-- end ngRepeat: crumb in breadcrumbs --><li ng-repeat=\\"crumb in breadcrumbs\\" ng-class=\\"['breadcrumb-item', { active: $last }]\\" aria-current=\\"page\\" class=\\"ng-scope breadcrumb-item active\\">
      <!-- ngIf: !$last -->
      <span ng-show=\\"$last\\" class=\\"ng-binding\\">Test 2</span>
    </li><!-- end ngRepeat: crumb in breadcrumbs -->
  </ol>
</nav>"
`;
